components/cors-mgt/org.wso2.carbon.identity.cors.mgt.core/src/main/java/org/wso2/carbon/identity/cors/mgt/core/internal/impl/CORSManagementServiceImpl.java

```diff
@@ -0,0 +1,333 @@
+                getConfigurationManager().addAttribute(CORS_ORIGIN_RESOURCE_TYPE_NAME, CORS_ORIGIN_RESOURCE_NAME,
+                        attribute);
+            }
+        } catch (ConfigurationManagementException | IOException e) {
+            throw handleServerException(ERROR_CODE_CORS_CRUD, e, "adding", tenantDomain);
+        } finally {
+            FrameworkUtils.endTenantFlow();
+        }
+    }
+
+    @Override
+    public void deleteCORSOrigins(String tenantDomain, List<CORSOrigin> corsOrigins)
+            throws CORSManagementServiceException {
+
+        validateTenantDomain(tenantDomain);
+        validateOrigins(corsOrigins);
+        try {
+            FrameworkUtils.startTenantFlow(tenantDomain);
+            addCORSOriginResourceTypeIfNotExists();
+
+            // Check if origins are not present
+            for (CORSOrigin corsOrigin : corsOrigins) {
+                if (!tenantHasCORSOrigin(corsOrigin)) {
+                    if (log.isDebugEnabled()) {
+                        log.debug(String.format(String.format(ERROR_CODE_ORIGIN_NOT_PRESENT.getMessage(),
+                                tenantDomain, corsOrigin)));
+                    }
+                    throw handleClientException(ERROR_CODE_ORIGIN_NOT_PRESENT, tenantDomain, corsOrigin.getUrl());
+                }
+            }
+
+            for (CORSOrigin corsOrigin : corsOrigins) {
+                Attribute attribute = new CORSOriginToAttribute().apply(corsOrigin);
+                getConfigurationManager().deleteAttribute(CORS_ORIGIN_RESOURCE_TYPE_NAME, CORS_ORIGIN_RESOURCE_NAME,
+                        attribute.getKey());
+            }
+        } catch (ConfigurationManagementException | IOException e) {
+            throw handleServerException(ERROR_CODE_CORS_CRUD, e, tenantDomain, "deleting");
+        } finally {
+            FrameworkUtils.endTenantFlow();
+        }
+    }
+
+    /**
+     * Add the CORS Origin resource type to the ConfigurationManager if not already present.
+     *
+     * @throws ConfigurationManagementException
+     */
+    private void addCORSOriginResourceTypeIfNotExists() throws ConfigurationManagementException {
+
+        if (isCORSOriginResourceTypeNotExists()) {
+            ResourceTypeAdd resourceTypeAdd = createCORSOriginResourceTypeToAdd();
+            getConfigurationManager().addResourceType(resourceTypeAdd);
+        }
+    }
+
+    /**
+     * Returns true if the CORS Origin type is already in the ConfigurationManager.
+     *
+     * @return {@code true} if the CORS Origin resource type is already in the ConfigurationManager,
+     * {@code false} otherwise.
+     * @throws ConfigurationManagementException
+     */
+    private boolean isCORSOriginResourceTypeNotExists() throws ConfigurationManagementException {
+
+        try {
+            getConfigurationManager().getResourceType(CORS_ORIGIN_RESOURCE_TYPE_NAME);
+        } catch (ConfigurationManagementClientException e) {
```

Comment:
Configuration manager should be ideally throwing a not found exception, rather than the cosumers checking for an error code. Shall we fix this. @mefarazath , @emswbandara , @ashensw thoughts ?

Reply:
for other similar OSGi services we followed the same approach. We defined a specific client error code and added a mapping to send 404 not found to the clients.
e.g. 
https://github.com/wso2/identity-rest-dispatcher/blob/master/components/org.wso2.carbon.identity.api.dispatcher/src/main/resources/ErrorMappings.properties#L29

Reply:
Is that the correct thing to do?
IMO, exception type should denote this other than further string processing

Reply:
Yes. IMO we should fix this behaviour.
Current behaviour makes it very hard to check for the existence of something in the ConfigurationManager as it requires the callee to handle exceptions and check error codes just to see if something exists.

Reply:
@ivantha please create an issue for this and point here.
We can look into that separately

Reply:
https://github.com/wso2/product-is/issues/8555